### DISCUSSION TITLE
Disable multitasking

### DISCUSSION THREAD

**Discussion by dhrumil83** [microsoft/Microsoft-Fabric-workload-development-sample#169] [Q&A] [answered]
How can we disable multi-tasking in the sample workload and get the default behavior?

Not seeing info in the docs: https://learn.microsoft.com/en-us/fabric/workload-development-kit/multitasking

Tried putting Block/Deny/false here: https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/tools/BaseTab.js#L4

But did not seem to work.

**Comment by shaym83** [comment, marked as answer]
Based on the behavior you want your workload to have, there are several options.
Your best approach would be setting allowMultipleTabs to "notAllow".
This will cause the item editor to be closed once you navigate away from it.

Another option is to set maxInstanceCount to 1 in [Item1.json](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/Package/Item1.json).
This will cause the item editor to remain open as a tab on the left nav bar but will not allow creation of new items (and thus new tabs) until that tab is closed.

**Reply by dhrumil83** [reply]
`notAllow` option works while running locally, but it defaults to it being enabled when deployed. 

maxInstanceCount set to 1 does not really work for our use case, since it gives this message when opening another item.

![Screenshot 2024-12-12 at 4 37 49 PM](https://github.com/user-attachments/assets/5ab9b9c6-2130-439d-9e0f-6d18b2aa4449)

**Reply by zainadr** [reply]
Multitasking is supported by default to all users with allowMultipleTabs is set to 'Allow' and maxInstanceCount is set to '1', user can only change **locally** to 'notAllow'. it means when deploying you cannot change allowMultipleTabs, you can only change the maxInstanceCount (up to 10).

**Reply by dhrumil83** [reply]
thanks @zainadr 
To clarify, we cannot turn off multiple tabs.
We can set it to 1, but in that case user will see the message: `Can't open the <xyz> item` when they already have 1 tab open.

**Reply by zainadr** [reply]
locally you can set allowMultipleTabs  to notAllow, else you cannot turn it off, you can only decide number of items that can be opened at the same time by setting maxInstanceCount property. for example, if you set maxInstanceCount to 'n', user can open 'n' items at the same time, when he tries to open the 'n+1' item, a message will pop up saying that 'Can't open more than 'n' items'.
